### PR TITLE
Track API calls per request and handle max fields per request

### PR DESCRIPTION
**Pull request by raman325**

### CODE REVIEW

**Comment by raman325** [comment]
@lymanepp can you review this one too before I merge? I have related upstream changes that I want to submit PRs for but would need to do another patch release and bump the version in HA before I do so

File: pytomorrowio/const.py
Comment (lymanepp) [inline pytomorrowio/const.py:24]: It's not possible to request more than `MAX_FIELDS` for any given `timestep` value. I'm adding a test case for that.

**Comment by lymanepp** [comment]
> @lymanepp can you review this one too before I merge? I have related upstream changes that I want to submit PRs for but would need to do another patch release and bump the version in HA before I do so

I'm done reviewing. You may not have seen my comments last time because they were just comments... not a real review.

**Review by lymanepp — CHANGES_REQUESTED** [review]

File: pytomorrowio/pytomorrowio.py
Comment (raman325) [reply pytomorrowio/pytomorrowio.py:291]: the point here is to support more than `MAX_FIELDS` by iterating through `MAX_FIELDS` at a time and collating the data. I have a dependent PR to add a number of new sensors that will cause us to exceed the limit

Comment (lymanepp) [reply pytomorrowio/pytomorrowio.py:291]: Gotcha

Comment (raman325) [reply pytomorrowio/pytomorrowio.py:291]: ah I see the concern - you are saying that `available_fields` returns less than 55 in all cases. I think I'm going to get rid of `available_fields`. When I initially started working on this, I wanted to encode some intelligence into the library about fields, but it's tedious and easier to handle upstream.

Comment (raman325) [reply pytomorrowio/pytomorrowio.py:291]: hmm I forgot about the measurements piece, so may need to hold off on this for now

Comment (lymanepp) [reply pytomorrowio/pytomorrowio.py:291]: It's a tricky problem. We definitely want to minimize the number of API calls for free API users (like me). But the difference in data required for hourly and daily forecasts pretty big imo.

Comment (raman325) [reply pytomorrowio/pytomorrowio.py:291]: can you give me some examples? It's been so long since I looked at all the different forecast types that it's no longer top of mind so I would have to relook it up

Comment (raman325) [reply pytomorrowio/pytomorrowio.py:291]: all of the attributes that `homeassistant` supports natively aren't all that different between the different forecast types. I guess down the road we may want to add more forecast attributes, but seems like a problem to tackle when we get there. For now, with the measurements we are using, we can use the same field list for both hourly and daily I think, but let me know if you think I am missing something

**Pull request closed without merging**
